assimilation_code/modules/observations/DEFAULT_obs_kind_mod.F90

```diff
@@ -1359,23 +427,21 @@ end function get_index_for_type_of_obs
 ! as well as the derived types - this will make the restart file
 ! handling more readable.
 
-function get_index_for_quantity(obs_kind_name)
+function get_index_for_quantity(obs_qty_name)
 
-! Returns the integer index corresponding to an observation kind string name
+! Returns the integer index corresponding to an observation quantity string name
 ! Returns a -1 if this string is not in list
 
-character(len=*), intent(in)  :: obs_kind_name
+character(len=*), intent(in)  :: obs_qty_name
 integer                       :: get_index_for_quantity
 
 integer :: i
-character(len=len(obs_kind_name)) :: string1
+character(len=*), parameter :: routine = 'get_index_for_quantity'
 
 if (.not. module_initialized) call initialize_module
 
-string1 = adjustl(obs_kind_name)
-
 do i = 0, max_defined_quantities
```

Comment:
here is one place a qty loop starts at 0.  i'm sure there are others in this file.